GVRf/Extensions/gvrf-physics/src/main/jni/engine/bullet/bullet_conetwistconstraint.cpp

```diff
@@ -0,0 +1,140 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// Created by c.bozzetto on 06/06/2017.
+//
+
+#include "bullet_conetwistconstraint.h"
+#include "bullet_rigidbody.h"
+
+#include <BulletDynamics/ConstraintSolver/btConeTwistConstraint.h>
+#include <LinearMath/btScalar.h>
+
+namespace gvr {
+    BulletConeTwistConstraint::BulletConeTwistConstraint(PhysicsRigidBody *rigidBodyB,
+                                                         PhysicsVec3 pivot,
+                                                         PhysicsMat3x3 const &bodyRotation,
+                                                         PhysicsMat3x3 const &coneRotation) {
+        mConeTwistConstraint = 0;
+        mRigidBodyB = reinterpret_cast<BulletRigidBody*>(rigidBodyB);
+
+        mBreakingImpulse = SIMD_INFINITY;
+        mPivot = pivot;
+        mBodyRotation = bodyRotation;
+        mConeRotation = coneRotation;
+        mSwingLimit = SIMD_PI * 0.25f;
+        mTwistLimit = SIMD_PI;
+    }
+
+    BulletConeTwistConstraint::~BulletConeTwistConstraint() {
+        if (0 != mConeTwistConstraint) {
+            delete mConeTwistConstraint;
+        }
+    }
+
+    void BulletConeTwistConstraint::setSwingLimit(float limit) {
+        if (0 != mConeTwistConstraint) {
+            mConeTwistConstraint->setLimit(4, limit);
+            mConeTwistConstraint->setLimit(5, limit);
```

Comment:
Bullet has two swing spam limits which allows one to set a "rollercoaster like" translation trajectory when these limits have different values. I assume this was not a valuable feature then I decided to set the same value to both swing spam limits.